File: src/main/java/scripting/map/FieldScriptManager.java
package scripting.map;

import client.Client;
import field.Field;
import scripting.AbstractScriptManager;

import javax.script.Invocable;

public class FieldScriptManager extends AbstractScriptManager {

    private static final FieldScriptManager instance = new FieldScriptManager();

    public static FieldScriptManager getInstance() {
        return instance;
    }

    private FieldScriptManager() {
    }

    public void execute(Client c, Field field, String script) {
        try {
            FieldScriptMethods map = new FieldScriptMethods(c, field);
            Invocable iv = getInvocable("map/" + script + ".js", c);
            if (iv == null) {
                System.out.println("Mapscript " + script + " is uncoded. (" + field.getId() + ")");
                return;
            }
            engine.put("field", map);
            iv.invokeFunction("execute");
        } catch (final Exception e) {
            e.printStackTrace();
        }
    }
}

File: src/main/java/scripting/map/FieldScriptManager.kt
package scripting.map

import client.Client
import field.Field
import scripting.AbstractScriptManager

object FieldScriptManager : AbstractScriptManager() {

    fun execute(c: Client, field: Field, script: String) {
        try {
            val map = FieldScriptMethods(c, field)
            val iv = getInvocable("map/$script.js", c)
            if (iv == null) {
                println("Mapscript " + script + " is uncoded. (" + field.id + ")")
                return
            }
            val unmutableEngine = engine ?: return
            unmutableEngine.put("field", map)
            iv.invokeFunction("execute")
        } catch (e: Exception) {
            e.printStackTrace()
        }
    }
}